assisted-events-scrape/workers/cluster_events_worker.py

```diff
@@ -113,6 +106,18 @@ def _internal_get_hosts():
             default_return_value=[]
         )
 
+    def _get_hosts_infraenvs(self, hosts):
+        hosts_infra_envs = []
+        try:
+            hosts_infra_envs = self.__get_infra_envs(
+                [host["infra_env_id"] for host in hosts
+                 if host and "infra_env_id" in host and host["infra_env_id"] is not None]
+            )
+            self.__set_infra_envs(hosts_infra_envs)
```

Comment:
it's a bit weird for a ``get`` method to have a side-effect
iiuc the code

Reply:
Yes, I agree it's a bit weird, we do it to make sure that we collect all infra env, both when we request list and single elements.